c_emulator/riscv_sim.cpp

```diff
@@ -366,9 +359,8 @@ static int process_args(int argc, char **argv)
       break;
 #ifdef RVFI_DII
     case 'r':
-      rvfi_dii = true;
       rvfi_dii_port = atoi(optarg);
-      fprintf(stderr, "using %d as RVFI port.\n", rvfi_dii_port);
+      rvfi = new rvfi_t(rvfi_dii_port);
```

Comment:
We could avoid the need for heap allocation and the need to free if we use std::optional<rvfi_t>.
```suggestion
      rvfi = rvfi_t(rvfi_dii_port);
```